diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ad9640..d4f56db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,9 +33,9 @@ find_package(GIO REQUIRED)
 
 find_package(Qt5DBus)
 
-find_package(Anthy)
-find_package(Pinyin)
-find_package(Chewing)
+# find_package(Anthy)
+# find_package(Pinyin)
+# find_package(Chewing)
 
 set(MALIIT_KEYBOARD_LIB_SOURCES
         src/lib/logic/abstractlanguagefeatures.h
@@ -357,7 +357,7 @@ install(FILES qml/constants.js qml/Keyboard.qml qml/KeyboardContainer.qml qml/Wo
 
 install(DIRECTORY data/styles
         DESTINATION ${MALIIT_KEYBOARD_DATA_DIR})
-install(FILES data/styles/ubuntu/images/icon_backspace18.png
+install(FILES data/styles/ubuntu/images/icon_backspace@18.png
         DESTINATION ${MALIIT_KEYBOARD_DATA_DIR}/icons
         RENAME erase.png)
 install(FILES data/styles/ubuntu/images/icon_enter@18.png
